sdk/nodejs/tests/runtime/langhost/run.spec.ts

```diff
@@ -1549,6 +1551,45 @@ describe("rpc", () => {
                 return { urn: makeUrn(t, name), id: undefined, props: undefined };
             },
         },
+        source_position: {
+            program: path.join(base, "074.source_position"),
+            expectResourceCount: 2,
+            registerResource: (
+                ctx: any,
+                dryrun: boolean,
+                t: string,
+                name: string,
+                res: any,
+                dependencies?: string[],
+                custom?: boolean,
+                protect?: boolean,
+                parent?: string,
+                provider?: string,
+                propertyDeps?: any,
+                ignoreChanges?: string[],
+                version?: string,
+                importID?: string,
+                replaceOnChanges?: string[],
+                providers?: any,
+                sourcePosition?: runtime.SourcePosition,
+            ) => {
+                assert(sourcePosition !== undefined);
+                assert(sourcePosition.uri.endsWith("index.js"));
+
+                switch (name) {
+                    case "custom":
+                        assert.strictEqual(sourcePosition.line, 2);
+                        break;
+                    case "component":
+                        assert.strictEqual(sourcePosition.line, 2);
+                        break;
```

Comment:
Why are these line 2? Isn't it 16 and 17?

Reply:
I am confused by this as well. I've had correct results from Typescript files but I haven't tried JS outside of the test harness. I'm wondering if this is related to the harness or if we'll always see this for JS.